test/support/acceptance.go

```diff
@@ -1,34 +1,125 @@
 package support
 
 import (
+	"encoding/json"
+	"gopkg.in/yaml.v3"
 	"log"
 	"os"
 	"path/filepath"
 	"regexp"
 	"strings"
 )
 
-func GetImageDefinitionsFromJson(jsonData map[string]interface{}) []string {
-	var images []string
-	for _, value := range jsonData {
-		switch v := value.(type) {
-		case map[string]interface{}:
-			images = append(images, GetImageDefinitionsFromJson(v)...)
-		case string:
-			if strings.HasPrefix(v, "quay.io") ||
-				strings.HasPrefix(v, "registry.redhat.io") ||
-				strings.Contains(v, "@sha256:") {
-				images = append(images, v)
-			}
+func GetImageDefinitionsFromSnapshot(jsonData Snapshot) []string {
+	return []string{
+		jsonData.CertificateTransparencyGo.CertificateTransparencyGo,
+		jsonData.Cli.ClientServerCg,
+		jsonData.Cli.ClientServerRe,
+		jsonData.Cli.Cosign,
+		jsonData.Cli.Gitsign,
+		jsonData.FbcV413.FbcV413,
+		jsonData.FbcV414.FbcV414,
+		jsonData.FbcV415.FbcV415,
+		jsonData.Fulcio.FulcioServer,
+		jsonData.Operator.RhtasOperator,
+		jsonData.Operator.RhtasOperatorBundle,
+		jsonData.Rekor.BackfillRedis,
+		jsonData.Rekor.RekorCli,
+		jsonData.Rekor.RekorServer,
+		jsonData.RekorSearchUI.RekorSearchUI,
+		jsonData.Scaffold.Createctconfig,
+		jsonData.Scaffold.CtlogManagectroots,
+		jsonData.Scaffold.FulcioCreatecerts,
+		jsonData.Scaffold.TrillianCreatedb,
+		jsonData.Scaffold.TrillianCreatetree,
+		jsonData.Scaffold.TufServer,
+		jsonData.SegmentBackupJob.SegmentBackupJob,
+		jsonData.Trillian.Database,
+		jsonData.Trillian.Logserver,
+		jsonData.Trillian.Logsigner,
+		jsonData.Trillian.Redis,
+	}
+}
+
+func GetCorrespondingSnapshotImage(operatorImageKey string, snapshotJsonData Snapshot) string {
```

Comment:
I believe, we can get rid of this static function. I found a pattern in matching names and I guess, we can get image values from struct using `"reflect"` package.

Reply:
Removed - parsing snapshot into simple map